diff --git a/core/includes/install.core.inc b/core/includes/install.core.inc
index 68ef6a2c81a..9c91f0355ad 100644
--- a/core/includes/install.core.inc
+++ b/core/includes/install.core.inc
@@ -652,7 +652,7 @@ function install_run_task($task, &$install_state) {
       // Otherwise, the batch will complete.
       // Disable the default script for the URL and clone the object, as
       // batch_process() will add additional options to the batch URL.
-      $url = Url::fromUri('base:install.php', ['query' => $install_state['parameters'], 'script' => '']);
+      $url = Url::fromUri('base:core/install.php', ['query' => $install_state['parameters'], 'script' => '']);
       $response = batch_process($url, clone $url);
       if ($response instanceof Response) {
         \Drupal::request()->getSession()->save();
