diff --git a/node_modules/php-cgi-wasm/PhpCgiBase.mjs b/node_modules/php-cgi-wasm/PhpCgiBase.mjs
index b5d9fbb..01f7b90 100644
--- a/node_modules/php-cgi-wasm/PhpCgiBase.mjs
+++ b/node_modules/php-cgi-wasm/PhpCgiBase.mjs
@@ -406,7 +406,7 @@ export class PhpCgiBase
 		else
 		{

-			path = docroot + '/' + rewrite.substr((vHostPrefix || this.prefix).length);
+			path = docroot + '/' + rewrite.substr((vHostPrefix || this.prefix).length).replace(/^\/+/, '');
 			scriptName = path;
 		}

@@ -419,6 +419,15 @@ export class PhpCgiBase

 		const extension = path.split('.').pop();

+		if(vHostEntrypoint)
+		{
+			if (extension === 'php') {
+				scriptName = vHostPrefix + '/' + rewrite.substr((vHostPrefix || this.prefix).length).replace(/^\/+/, '')
+			} else {
+				scriptName = vHostPrefix + '/' + vHostEntrypoint;
+			}
+		}
+
 		if(extension !== 'php' && extension !== 'phar')
 		{
 			const aboutPath = php.FS.analyzePath(path);
